pkg/api/login.go

```diff
@@ -135,7 +144,16 @@ func (hs *HTTPServer) LoginPost(c *models.ReqContext, cmd dtos.LoginCommand) Res
 
 	if redirectTo, _ := url.QueryUnescape(c.GetCookie("redirect_to")); len(redirectTo) > 0 {
 		result["redirectUrl"] = redirectTo
-		c.SetCookie("redirect_to", "", -1, setting.AppSubUrl+"/")
+		c.Resp.Header().Del("Set-Cookie")
```

Comment:
You cannot delete the "Set-Cookie" header since `hs.loginUserWithUser` above sets the auth cookie using `http.SetCookie`. If you delete the "Set-Cookie" header here you basically logging out the user.

Reply:
Make sense, I've deleted the 'Set-Cookie' delete in the code. Thanks for noticing.